masmini-common/src/main/java/masmini/Resource.kt

```diff
@@ -34,80 +32,37 @@ open class Resource<out T> @PublishedApi internal constructor(val value: Any?) {
      * Get the current value if successful, or null for other cases.
      */
     fun getOrNull(): T? =
-        when {
-            isLoading -> (value as Loading<T>).value
-            isSuccess -> value as T?
-            else -> null
-        }
+            when {
+                isSuccess -> value as T?
+                else -> null
+            }
 
     fun exceptionOrNull(): Throwable? =
-        when (value) {
-            is Failure -> value.exception
-            else -> null
-        }
+            when (value) {
+                is Failure -> value.exception
+                else -> null
+            }
 
     companion object {
         fun <T> success(value: T): Resource<T> = Resource(value)
         fun <T> failure(exception: Throwable? = null): Resource<T> = Resource(Failure(exception))
         fun <T> loading(value: T? = null): Resource<T> = Resource(Loading(value))
-        fun <T> empty(): Resource<T> = Resource(Empty)
-    }
-
-    override fun toString(): String {
-        return if (isSuccess) "Success($value)" else value.toString()
-    }
-
-    override fun equals(other: Any?): Boolean {
-        if (this === other) return true
-        if (javaClass != other?.javaClass) return false
-
-        other as Resource<*>
-
-        if (value != other.value) return false
-
-        return true
-    }
-
-    override fun hashCode(): Int = value?.hashCode() ?: 0
-}
+        fun <T> empty(): Resource<T> = Resource(Empty())
 
-/**
- * A resource that abstracts asynchronous operation but with idle
- * state instead of empty.
- *
- * It accepts temporary metadata objects as value.
- */
-open class TypedTask<T> @PublishedApi internal constructor(value: Any?) : Resource<T>(value) {
```

Comment:
We should review if we recover this so we don't break compatibility. We used it to store metadata related to the task that doesn't make sense to store in a state since the metadata may be volatile